### ISSUE TITLE
Bug: Sort By Size Not Working

### ISSUE THREAD

**Issue by zero77** [ArchiveBox/ArchiveBox#1495]
#### Describe the bug
Under snapshot when you click on size either once or twice it does not sort by size but does reorder the list of snapshots.

#### Steps to reproduce
Go to snapshot click on size either once or twice 

#### ArchiveBox version
v0.8.2

**Comment by pirate** [comment]
That is correct, size is not stored in DB as it's an expensive filesystem calculation, it's computed on the fly and cached only for the snapshots currently visible on the page. When you sort by size it actually sorts by number of output results recorded, which is the closest approximation that's still reasonably fast.

If you really need to sort by size I suggest just sorting the data/archive/ dir in your filesystem browser or using a CLI tool like Ncdu (both of which you'll notice take several minutes to compute sizes for everything).

**Comment by pirate** [comment]
The displayed sizes are accurate but it's doing a trick where it only calculates them for the page you're actively viewing (<100 snapshots at a time). The sorting needs to operate across *all pages* matching your active filters, so it cant be based on the displayed sizes in the current architecture without having to calculate the size of every single snapshot in the entire dataset.
